[small] Consolidate/deduplicate some Framework Move structs in rust

igor-aptos opened the pull request:
## Description
We have some of the common structs in multiple places - Rosetta, tests. I needed to add some of these for perf testing as well, so consolidating them a bit first.

## How Has This Been Tested?
CI

## Type of Change
- [x] Refactoring

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

<!-- Thank you for your contribution! -->

lightmark reviewed (approved):
lgtm
@gregnazario take a look on rosetta?

igor-aptos commented:
@gregnazario landing, as changes should be trivial, but if you have any concerns, let me know.

igor-aptos commented:
@lightmark , @msmouse fyi - made file layout more consistent (all fa events in one file, all coin events in another), and added MoveEventV1Type and TypeInfoResource.

fixed the unit test (there was a typo in the strings). landing now
